src/Window.cpp

```diff
@@ -1299,8 +1339,10 @@ bool Window::wallLeft() {
     return isWall({position.first, position.second, direction});
 }
 
-bool Window::moveForward() {
-    if (wallFront()) {
+bool Window::moveForward(int distance) {
+    moveLength = distance;
+    // check for crashes and negative distances
+    if (distance < 0 || wallFront(true)) { // the true parameter tells wallFront() to look ahead for moveLength tiles
```

Comment:
As written, the simulator will immediately respond `crash` if the mouse can't move forward the request number of spaces. I wonder... would it be crazy for the mouse to move forward as many spaces as possible *and then* crash? That'd be a tad more realistic, right? If that sounds reasonable to you, you should store `m_forwardMovesRemaining` and decrement by one after each successful movement forward. If you hit zero, you can return `ACK`. If you encounter a wall before hitting zero, you can return `CRASH`.

Reply:
I agree that this is more realistic. However, I'm not sure how/where to implement this. Currently, the way `processQueuedCommands()` is written, the mouse cannot move at all after a command that results in "crash".

Reply:
Yea, good point, it'd be a bit more complicated. I think it's fine to leave the logic as is for now.

However, rather than storing `distance` in an instance variable and then passing `true` to `wallFront`, how about just passing `distance` into `wallFront`?

Reply:
I'll make the change right now to pass `distance` to wallFront. However, the instance variable `moveLength` is still in use in the function `updateMouseProgress()`. Is that alright?

Reply:
Oh, right, totally forgot. Yea, that's fine :).